app.go

```diff
@@ -607,9 +618,12 @@ func (a *Application) processTx(txBytes []byte, isCheckTx bool) (TxHandlerResult
 	receiptHandler := a.ReceiptHandlerProvider.Store()
 	r, err := a.TxHandler.ProcessTx(state, txBytes, isCheckTx)
 	if err != nil {
+		if !isCheckTx && state.FeatureEnabled(EvmTxReceiptsVersion2_1Feature, false) {
```

Comment:
The receipts are saved to receipts_db and don't affect the app hash anymore, so the feature flag shouldn't be necessary right?

Reply:
Right, we can just change it. Feature flag removed.